src/PointwiseFunctions/GeneralRelativity/WeylElectric.cpp

```diff
@@ -47,30 +47,98 @@ void weyl_electric(
     }
   }
 }
+
+namespace {
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_electric_scalar_impl(
+    const gsl::not_null<Scalar<DataType>*> weyl_electric_scalar_result,
+    const gsl::not_null<tnsr::Ij<DataType, SpatialDim, Frame>*>
+        weyl_electric_up_down,
+    const tnsr::ii<DataType, SpatialDim, Frame>& weyl_electric,
+    const tnsr::II<DataType, SpatialDim, Frame>&
+        inverse_spatial_metric) noexcept {
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t k = 0; k < SpatialDim; ++k) {
+        weyl_electric_up_down->get(j, k) +=
+            weyl_electric.get(i, k) * inverse_spatial_metric.get(i, j);
+      }
+    }
+  }
+  for (size_t j = 0; j < SpatialDim; ++j) {
+    for (size_t k = 0; k < SpatialDim; ++k) {
+      if (UNLIKELY(j == 0 and k == 0)) {
+        get(*weyl_electric_scalar_result) =
+            weyl_electric_up_down->get(j, k) * weyl_electric_up_down->get(k, j);
+      } else {
+        get(*weyl_electric_scalar_result) +=
+            weyl_electric_up_down->get(j, k) * weyl_electric_up_down->get(k, j);
+      }
+    }
+  }
+}
+}  // namespace
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_electric_scalar(
+    const gsl::not_null<Scalar<DataType>*> weyl_electric_scalar_result,
+    const tnsr::ii<DataType, SpatialDim, Frame>& weyl_electric,
+    const tnsr::II<DataType, SpatialDim, Frame>&
+        inverse_spatial_metric) noexcept {
+  *weyl_electric_scalar_result =
+      make_with_value<Scalar<DataType>>(get<0, 0>(inverse_spatial_metric), 0.0);
+
+  auto weyl_electric_up_down =
+      make_with_value<tnsr::Ij<DataType, SpatialDim, Frame>>(
+          get<0, 0>(inverse_spatial_metric), 0.0);
+  weyl_electric_scalar_impl(weyl_electric_scalar_result,
```

Comment:
I don't see the need for `weyl_electric_scalar_impl`; I would just put the body of that function here; then you don't need to treat `weyl_electric_up_down` as a pointer.

Reply:
ping on this